### PR TITLE
Remove query string in URI when converting GET to POST

### PR DESCRIPTION
**Pull request by cjyclaire**
This fix remove query string in URI when converting GET to POST.
Fix issue #1047 

cc: @chrisradek @LiuJoyceC

### CODE REVIEW

File: tests/CloudSearchDomain/CloudSearchDomainTest.php [hunk 1]
```diff
@@ -45,5 +45,6 @@ public function testConvertGetToPost()
         $this->assertEquals('application/x-www-form-urlencoded', $request->getHeaderLine('Content-Type'));
         $this->assertEquals(7, $request->getHeaderLine('Content-Length'));
         $this->assertEquals('foo=bar', $request->getBody());
+        $this->assertEquals('', $request->getUri()->getQuery());
```
Comment (GrahamCampbell) [inline tests/CloudSearchDomain/CloudSearchDomainTest.php]: Maybe use `assertSame` otherwise things like `0` and `null` pass this assertion.

**Comment by chrisradek** [comment]
LGTM

File: tests/CloudSearchDomain/CloudSearchDomainTest.php (hunk 1 above)
Comment (cjyclaire) [inline tests/CloudSearchDomain/CloudSearchDomainTest.php]: Make sense, done in latest commit.

**Comment by LiuJoyceC** [comment]
🚢

**Pull request merged**